src/lib/NumberFormatting.js

```diff
@@ -12,9 +12,18 @@ class NumberFormatting { // eslint-disable-line no-unused-vars
         const roundingFactor = 10 ** maxDecimals;
         value = Math.round(value * roundingFactor) / roundingFactor;
 
-        const result = parseFloat(value.toFixed(minDecimals)) === value
+        let result = parseFloat(value.toFixed(minDecimals)) === value
             ? value.toFixed(minDecimals)
-            : value.toString();
+            : value.toFixed(maxDecimals);
+
+        // Remove trailing zeros until minDecimals has been reached
+        while (result.split('.')[1] && result.split('.')[1].length > minDecimals) {
+            if (result[result.length - 1] === '0') {
+                result = result.substring(0, result.length - 1);
+            } else {
+                break;
+            }
+        }
```

Comment:
```suggestion
        while (result.split('.')[1] && result.split('.')[1].length > minDecimals && result[result.length - 1] === '0') {
            result = result.replace(/\.?0$/, '');
        }
```
Simplify the loop and avoid potential dangling decimal separator.